tests/src/test/java/com/cloudera/director/google/sql/GoogleCloudSQLProviderConfigurationValidatorTest.java

```diff
@@ -0,0 +1,185 @@
+  }
+
+  /**
+   * Invokes checkRegion with the specified configuration.
+   *
+   * @param region the region name
+   */
+  protected void checkRegion(String region) {
+    Map<String, String> configMap = Maps.newHashMap();
+    configMap.put(REGION_SQL.unwrap().getConfigKey(), region);
+    Configured configuration = new SimpleConfiguration(configMap);
+    validator.checkRegion(configuration, accumulator, localizationContext);
+  }
+
+  /**
+   * Verifies that the specified plugin exception condition accumulator contains no errors or
+   * warnings.
+   */
+  private void verifyClean() {
+    Map<String, Collection<PluginExceptionCondition>> conditionsByKey = accumulator.getConditionsByKey();
+    assertThat(conditionsByKey).isEmpty();
+  }
+
+  /**
+   * Verifies that the specified plugin exception condition accumulator contains exactly
+   * one condition, which must be an error with the specified message and associated with the
+   * specified property.
+   *
+   * @param token    the configuration property token for the property which should be in error
+   * @param errorMsg the expected error message
+   * @param args     the error message arguments
+   */
+  private void verifySingleError(ConfigurationPropertyToken token, String errorMsg, Object... args) {
+    verifySingleError(token, Optional.of(errorMsg), args);
+  }
+
+  /**
+   * Verifies that the specified plugin exception condition accumulator contains exactly
+   * one condition, which must be an error with the specified message and associated with the
+   * specified property.
+   *
+   * @param token          the configuration property token for the property which should be in error
+   * @param errorMsgFormat the expected error message
+   * @param args           the error message arguments
+   */
+  private void verifySingleError(ConfigurationPropertyToken token, Optional<String> errorMsgFormat, Object... args) {
+    Map<String, Collection<PluginExceptionCondition>> conditionsByKey = accumulator.getConditionsByKey();
+    assertThat(conditionsByKey).hasSize(1);
+    String configKey = token.unwrap().getConfigKey();
+    assertThat(conditionsByKey.containsKey(configKey)).isTrue();
+    Collection<PluginExceptionCondition> keyConditions = conditionsByKey.get(configKey);
+    assertThat(keyConditions).hasSize(1);
+    PluginExceptionCondition condition = keyConditions.iterator().next();
+    verifySingleErrorCondition(condition, errorMsgFormat, args);
+  }
+
+  /**
+   * Verifies that the specified plugin exception condition is an error with the specified message.
+   *
+   * @param condition      the plugin exception condition
+   * @param errorMsgFormat the expected error message format
+   * @param args           the error message arguments
+   */
+  private void verifySingleErrorCondition(PluginExceptionCondition condition, Optional<String> errorMsgFormat,
+      Object... args) {
+    assertThat(condition.isError()).isTrue();
```

Comment:
calling `assertTrue` requires some additional imports that I could not easily find.